### DISCUSSION TITLE
Using SettingsService in Static mode (not logged in)

### DISCUSSION THREAD

**Discussion by mdmontesinos** [oqtane/oqtane.framework#4143] [Q&A]
I've been trying to dynamically update the SiteSettings from my static modules in the OnInitialized method, but currently it's not possible due to the required authorization.

For example, I want to dynamically set a subheader in the theme depending on the module I am, so the easiest way I've thought of is by using the SiteSettings.

This would be the code in my Static module to update the settings:

```c#
Guid guid = Guid.NewGuid();
var settings = await SettingService.GetSiteSettingsAsync(PageState.Site.SiteId);
settings = SettingService.SetSetting(settings, "Subheader", guid.ToString());
await SettingService.UpdateSiteSettingsAsync(settings, PageState.Site.SiteId);
```

And I'd recover and display it in my theme:

```c#
var settings = SettingService.MergeSettings(PageState.Site.Settings, PageState.Page.Settings);
_login = bool.Parse(SettingService.GetSetting(settings, GetType().Namespace + ":Login", "true"));
_register = bool.Parse(SettingService.GetSetting(settings, GetType().Namespace + ":Register", "true"));
subheader = SettingService.GetSetting(settings, "Subheader", string.Empty);
```

However, this does not work because the SettingController requires and authorized user to perform this operation: "User Not Authorized To Update Setting {Setting}".

Is there any other alternative to set some global data in static mode (apart from query or path variables)?

**Comment by sbwalker** [comment]
@mdmontesinos There are 2 types of Site Settings - Private and Public. If you look in the database, these are controlled by an IsPrivate boolean field:

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/c0a16e1d-8b89-4944-acc3-ad052a2481ad)

Site Settings which are Public (IsPrivate = 0) do not require any authorization - they are "publicly" accessible through the API. Site Settings which are Private (IsPrivate = 1) require authorization and will not be shared through the API unless the user making the request has the necessary permissions. This is how Site Settings such as SMTP Configuration, etc... are kept confidential on the server.

In regards to updating you are correct that the API does not allow an anonymous user to update a Site Setting - as updating settings on a backend server should always be treated as a privileged operation. I am not sure what you are trying to accomplish, but if you truly need to allow an anonymous user to update a value in your backend database, then I would suggest possibly using a different mechanism. 

Oqtane does support Visitor Settings which are based on cookies and can be used to maintain settings in a database.

You could also use the INotifyPropertyChanged to pass information between components at run-time: https://www.oqtane.org/blog/!/40/inotifypropertychanged
